substrate/frame/staking-async/rc-client/src/lib.rs

```diff
@@ -381,30 +466,35 @@ where
 	/// split it into smaller pieces if XCM validation fails with `ExceedsMaxMessageSize`. It will
 	/// fail on other errors.
 	///
-	/// It will only emit some logs, and has no return value. This is used in the runtime, so it
-	/// cannot deposit any events at this level.
-	pub fn split_then_send(message: Message, maybe_max_steps: Option<u32>) {
+	/// Returns `Ok()` if the message was sent using `XCM`, potentially with splitting up to
+	/// `maybe_max_step` times, `Err(())` otherwise.
+	pub fn split_then_send(message: Message, maybe_max_steps: Option<u32>) -> Result<(), ()> {
 		let message_type_name = core::any::type_name::<Message>();
 		let dest = Destination::get();
-		let xcms = match Self::prepare(message, maybe_max_steps) {
-			Ok(x) => x,
-			Err(e) => {
-				log::error!(target: "runtime::rc-client", "📨 Failed to split message {}: {:?}", message_type_name, e);
-				return;
-			},
-		};
-
-		for (idx, xcm) in xcms.into_iter().enumerate() {
-			log::debug!(target: "runtime::rc-client", "📨 sending {} message index {}, size: {:?}", message_type_name, idx, xcm.encoded_size());
-			let result = send_xcm::<Sender>(dest.clone(), xcm);
-			match result {
-				Ok(_) => {
-					log::debug!(target: "runtime::rc-client", "📨 Successfully sent {} message part {} to relay chain", message_type_name,  idx)
-				},
-				Err(e) => {
-					log::error!(target: "runtime::rc-client", "📨 Failed to send {} message to relay chain: {:?}", message_type_name, e)
-				},
+		let xcms = Self::prepare(message, maybe_max_steps).map_err(|e| {
+			log::error!(target: "runtime::staking-async::rc-client", "📨 Failed to split message {}: {:?}", message_type_name, e);
+		})?;
+
+		match with_transaction_opaque_err(|| {
```

Comment:
@franciscoaguirre @bkontur I don't expect us to need this mechanism anymore, but as far as I can say this code should still be correct. Can you confirm?